crypto/src/action/output.rs

```diff
@@ -38,14 +92,69 @@ impl Body {
         let note = Note::new(dest, value, note_blinding);
         let note_commitment = note.commit();
         // TODO: Encrypt note here and add to a field in the Body struct (later).
+        // TEMP
+        let encrypted_note = [0u8; NOTE_ENCRYPTION_BYTES];
 
         let esk = ka::Secret::new(rng);
         let ephemeral_key = esk.diversified_public(note.dest.diversified_generator());
 
+        let proof = OutputProof {};
+
         Self {
             value_commitment,
             note_commitment,
             ephemeral_key,
+            encrypted_note,
+            proof,
         }
     }
 }
+
+impl Protobuf<transaction::OutputBody> for Body {}
+
+impl From<Body> for transaction::OutputBody {
+    fn from(msg: Body) -> Self {
+        let cv_bytes: [u8; 32] = msg.value_commitment.into();
+        let cm_bytes: [u8; 32] = msg.note_commitment.into();
+        let proof_bytes: [u8; OUTPUT_PROOF_LEN_BYTES] = msg.proof.into();
+        transaction::OutputBody {
+            cv: Bytes::copy_from_slice(&cv_bytes),
+            cm: Bytes::copy_from_slice(&cm_bytes),
+            ephemeral_key: Bytes::copy_from_slice(&msg.ephemeral_key.0),
+            encrypted_note: Bytes::copy_from_slice(&msg.encrypted_note),
+            zkproof: Bytes::copy_from_slice(&proof_bytes),
+        }
+    }
+}
+
+impl TryFrom<transaction::OutputBody> for Body {
+    type Error = Error;
+
+    fn try_from(proto: transaction::OutputBody) -> Result<Self, Self::Error> {
+        let value_commitment: value::Commitment = (proto.cv[..])
```

Comment:
since these types are specified by the fields of `Body`, the type ascription isn't necessary to constrain the `try_into()` call, and they could even be inlined into the `Body { ... }` expression